data/data/azure/bootstrap/main.tf

```diff
@@ -168,3 +168,27 @@ resource "azurerm_virtual_machine" "bootstrap" {
   }
 }
 
+resource "azurerm_network_security_group" "bootstrap" {
+  name                = "${var.cluster_id}-bootstrap-nsg"
+  location            = var.region
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_subnet_network_security_group_association" "bootstrap" {
+  subnet_id                 = azurerm_subnet.master_subnet.id
```

Comment:
you probably want to use `var.subnet_id` instead of the `azurerm_subnet.master_subnet.id`.

in terraform you only have access to resource in the same module, and you use variables and outputs of modules to share information between modules.

Reply:
Fixed!